test/Strings/replace-xsite.js

```diff
@@ -0,0 +1,21 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+    {
+        name: "String.Replace ( Pattern as String , ReplaceFunction )",
+        body: function () {
+            assert.doesNotThrow(function() {
+                var str = "AAoooAAooo";
+                var replacer = WScript.LoadScriptFile('replace-xsite-replacer.js', 'samethread')

```

Comment:
you can call WScript.LoadScript('source code'), instead of LoadScriptFile - that way you can avoid creating one more file.